Core.ApplicationServices/Organizations/OrganizationUnitService.cs

```diff
@@ -0,0 +1,345 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Core.Abstractions.Types;
+using Core.ApplicationServices.Authorization;
+using Core.ApplicationServices.Contract;
+using Core.ApplicationServices.Model.Organizations;
+using Core.ApplicationServices.SystemUsage;
+using Core.DomainModel.Organization;
+using Core.DomainServices.Authorization;
+using Core.DomainServices.Generic;
+using Infrastructure.Services.DataAccess;
+
+namespace Core.ApplicationServices.Organizations
+{
+    public class OrganizationUnitService : IOrganizationUnitService
+    {
+        private readonly IEntityIdentityResolver _identityResolver;
+        private readonly IOrganizationService _organizationService;
+        private readonly IOrganizationRightsService _organizationRightsService;
+        private readonly IItContractService _contractService;
+        private readonly IItSystemUsageService _usageService;
+        private readonly IAuthorizationContext _authorizationContext;
+        private readonly ITransactionManager _transactionManager;
+
+        public OrganizationUnitService(IEntityIdentityResolver identityResolver, 
+            IOrganizationService organizationService,
+            IOrganizationRightsService organizationRightsService, 
+            IItContractService contractService,
+            IItSystemUsageService usageService, 
+            IAuthorizationContext authorizationContext, 
+            ITransactionManager transactionManager)
+        {
+            _organizationService = organizationService;
+            _organizationRightsService = organizationRightsService;
+            _contractService = contractService;
+            _usageService = usageService;
+            _authorizationContext = authorizationContext;
+            _transactionManager = transactionManager;
+            _identityResolver = identityResolver;
+        }
+
+        public Result<OrganizationRegistrationDetails, OperationError> GetOrganizationRegistrations(int organizationId, int unitId)
+        {
+            return GetOrganization(organizationId, OrganizationDataReadAccessLevel.All)
+                .Match
+                (
+                    _ => GetOrganziationUnit(unitId),
+                    error => error
+                )
+                .Match
+                (
+                    unit => _authorizationContext.AllowReads(unit) == false
```

Comment:
no need to re-check.. you just checked that you have ALL access to the organization